### ISSUE TITLE
Show hint what to add to discriminator map for partial discriminator maps

### ISSUE THREAD

**Issue by rieschl** [doctrine/mongodb-odm#2032]
### Feature Request

`1.3-RC1` introduced deprecation warnings for the upcoming version `2.0`. One of them is a warning when a partial discriminator map is found ([e.g. here](https://github.com/doctrine/mongodb-odm/blob/1.3.x/lib/Doctrine/ODM/MongoDB/Persisters/PersistenceBuilder.php#L308)).

As discussed on Slack, , it would be nice if this warning included what should be added to the discriminator map to stay backwards compatible, i.e. it should output the FQCN of the class, exactly what is currently being added automatically. If I have existing documents with the automatically generated discriminators, I have to use the same value, because otherwise those documents would break.

**Comment by alcaeus** [comment]
I finally managed to take a look at this today. The three exception cases all affect the class that is being printed in the deprecation message, or its inheritance chain. Due to the way mapping is set up, we don't know what class defines the discriminator mapping, so it's best to check the class that is printed and go up the inheritance chain if it doesn't have a discriminator map.

While looking at this, I noticed that we didn't change this for one case: associations without a target document can optionally have a discriminator map, which currently can be incomplete. I'll have to add a deprecation warning there as well, but I have to think about how to get full information there. We currently only know the field name of the association, but not the class it's contained in. Since this is public API, I have to find a backwards compatible way of passing the information to the method.
